IdentityCore/src/network/request/MSIDAADAuthorityMetadataResponseSerializer.m

```diff
@@ -51,41 +51,32 @@ - (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse
         return nil;
     }
     
-    if ([jsonObject msidAssertContainsField:@"error" context:context error:nil]
-        && [jsonObject msidAssertType:NSString.class ofField:@"error" context:context errorCode:MSIDErrorServerInvalidResponse error:nil])
+    NSString *oauthError = [jsonObject msidStringObjectForKey:MSID_OAUTH2_ERROR];
```

Comment:
very nit: new code doesn't differentiate between error being not present and error being not of correct type anymore. We should at least log a warning if we get error of incorrect type.

Reply:
Added a warning.